data-prepper-api/src/main/java/org/opensearch/dataprepper/model/sink/SinkForwardRecordsContext.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.sink;
+
+public class SinkForwardRecordsContext {
+    public SinkForwardRecordsContext() {}
```

Comment:
Perhaps adding in here would make sense. What if a sink only writes some records to the destination?

This context would collect all of the successful ones.